WordPress/gutenberg
# Refactor, document, and fix image block deprecations

## What?


Cleans up the image block deprecations and fixes some of the attributes/supports for older versions.

## Why?


I found that some of the image block deprecations were wrong because block attributes and supports were being reused.

Naming the deprecations by version with documentation, I think, is easier to understand. It's also the recommended way to add deprecations in [the documentation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/#example).

I figure it's better to just copy and paste the current state of the block without relying on variables so folks don't accidentally update them again in the future.

## How?


1. Refactor to name each version `v1`, `v2`, etc.
2. Remove blockAttributes and blockSupports variables and replace with the accurate versions for when the block deprecation was added.
3. Add description and link to the PR adding the deprecation for each deprecation.
4. Fix integration tests that were incorrect.
5. Rename integration test fixtures to appear in the order of their deprecation and provide a short description of what the deprecation is testing.

## Testing Instructions





Should be covered by the image block fixtures tests, but you could also manually create a block for each version and see that they accurately get updated to the latest version.

### Testing Instructions for Keyboard


N/A

## Screenshots or screencast 

N/A
